app/gui2/shared/ast/tree.ts

```diff
@@ -2010,6 +2052,14 @@ function getAll<Fields extends object>(map: FixedMapView<Fields>): Fields {
   return Object.fromEntries(map.entries()) as Fields
 }
 
+declare const brandLegalFieldContent: unique symbol
+/** Used to add a constraint to all `AstFields`s subtypes ensuring that they were produced by `composeFieldData`, which
+ *  enforces a requirement that the provided fields extend `FieldObject`.
+ */
+interface LegalFieldContent {
+  [brandLegalFieldContent]: never
+}
+
 /** Modifies the input `map`. Returns the same object with an extended type. */
 export function setAll<Fields1, Fields2 extends object>(
```

Comment:
Let's use slightly narrower `Fields2 extends Record<string, any>` instead of `object`.